frontend/src/routes/manage/Video/VideoDetails.tsx

```diff
@@ -0,0 +1,101 @@
+import CONFIG from "../../../config";
+import {
+    UpdatedCreatedInfo,
+    DetailsPage,
+    DirectLink,
+    MetadataSection,
+    DeleteButton,
+    ButtonSection,
+    HostRealms,
+} from "../Shared/Details";
+
+
+export const ManageVideoDetailsRoute = makeManageVideoRoute(
+    "details",
+    "",
+    authEvent => <DetailsPage
+        pageTitle="manage.my-videos.details.title"
+        item={{ ...authEvent, updated: authEvent.syncedData?.updated }}
+        breadcrumb={{
+            label: i18n.t("manage.my-videos.title"),
+            link: ManageVideosRoute.url,
+        }}
+        sections={event => [
+            <UpdatedCreatedInfo key="created-info" item={{
+                ...event,
+                updated: event.syncedData?.updated,
+            }} />,
+            <VideoButtonSection key="button-section" event={authEvent} />,
+            <DirectLink key="direct-link" withTimestamp url={
+                new URL(DirectVideoRoute.url({ videoId: authEvent.id }), document.baseURI)
+            } />,
+            <MetadataSection key="metadata" item={event} />,
+            <div key="host-realms" css={{ marginBottom: 32 }}>
+                <HostRealms kind="videos" hostRealms={authEvent.hostRealms} itemLink={realmPath => (
+                    <Link to={VideoRoute.url({ realmPath: realmPath, videoID: authEvent.id })}>
+                        {i18n.t("video.video")}
+                    </Link>
+                )}/>
+            </div>,
+        ]}
+    />
+);
+
+const deleteVideoMutation = graphql`
+    mutation VideoDetailsDeleteMutation($id: ID!) {
+        deleteVideo(id: $id) { id }
+    }
+`;
+
+const VideoButtonSection: React.FC<{ event: AuthorizedEvent }> = ({ event }) => {
+    const { t, i18n } = useTranslation();
+    const [commit] = useMutation(deleteVideoMutation);
+    const user = useUser();
+    const config = useAppkitConfig();
+    const { isHighContrast } = useColorScheme();
+
+    if (!isRealUser(user)) {
+        return <NotAuthorized />;
+    }
+
+    return <ButtonSection>
+        {user.canUseEditor && !event.isLive && event.canWrite && (
+            <ExternalLink
+                service="EDITOR"
+                params={{
+                    id: event.opencastId,
+                    callbackUrl: document.location.href,
+                    callbackSystem: translatedConfig(CONFIG.siteTitle, i18n),
+                }}
+                fallback="button"
+                css={buttonStyle(config, "normal", isHighContrast)}
+            >
+                {t("manage.my-videos.details.open-in-editor")}
+            </ExternalLink>
+        )}
+        <DeleteButton
+            itemId={event.id}
+            itemTitle={event.title}
+            itemType="video"
+            returnPath="/~manage/videos"
+            commit={config => {
+                const disposable = commit(config);
+                return { [Symbol.dispose]: () => disposable.dispose() };
+            }}
```

Comment:
This is a bit weird, `commit` should already return a disposable, I'm sure you are aware. I have to look a bit closer at this to understand this fully...

Reply:
I made a dumb mistake. 🤦  Feel free to ignore the explanation.

The type for this `commit` prop is `commit: (config: UseMutationConfig<TMutation>) => Disposable;`.
Soo it turns out that `Disposable` is also an interface in the `@types` module, which is inferred if not explicitly imported from `relay-runtime`.

So, yeah. Doing that makes passing `commit={commit}` perfectly fine.